torch/_prims/rng_prims.py

```diff
@@ -144,6 +144,8 @@ def get_device(args, kwargs):
         return "cuda"
     elif any(dev == "cpu" for dev in devices):
         return "cpu"
+    elif any(dev == "hpu" for dev in devices):
+        return "hpu"
```

Comment:
You should put "hpu" check before "cpu" check so that even if there is any "cpu" tensors, we return "hpu" as long as there is any "hpu" tensor.

Reply:
good catch:)